integration-tests/smoke/ccip/ccip_messaging_test.go

```diff
@@ -603,6 +605,235 @@ func Test_CCIPMessaging_EVM2Solana(t *testing.T) {
+	/////////////////////////////////////////////////////////////////////////////////
+
+	// Set reject all flag in receiver to force reverts
+	ctx := testhelpers.Context(t)
+	solChains := e.Env.BlockChains.SolanaChains()
+	deployer := solChains[destChain].DeployerKey
+	ix, err := test_ccip_receiver.NewSetRejectAllInstruction(true, receiverTargetAccountPDA, deployer.PublicKey()).ValidateAndBuild()
+	require.NoError(t, err)
+	ixData, err := ix.Data()
+	require.NoError(t, err)
+	rejectAllIx := solana.NewInstruction(receiverProgram, ix.Accounts(), ixData)
+	res := soltestutils.SendAndConfirm(ctx, t, solChains[destChain].Client, []solana.Instruction{rejectAllIx}, *deployer, solconfig.DefaultCommitment)
+	require.Nil(t, res.Meta.Err)
+
+	t.Run("failed messages should only execute once", func(t *testing.T) {
+		// This test didn't finish, it timed out waiting for exec state change... not sure why when
+		// there is no expected status.
+		accounts := [][32]byte{
+			receiverExternalExecutionConfigPDA,
+			receiverTargetAccountPDA,
+			solana.SystemProgramID,
+		}
+
+		extraArgs, err := ccipevm.SerializeClientSVMExtraArgsV1(msg_hasher163.ClientSVMExtraArgsV1{
+			AccountIsWritableBitmap:  solccip.GenerateBitMapForIndexes([]int{0, 1}),
+			Accounts:                 accounts,
+			ComputeUnits:             1_000_000,
+			AllowOutOfOrderExecution: true,
+		})
+		require.NoError(t, err)
+
+		/*
+			deployer := solChains[destChain].DeployerKey
+			// Set reject all flag in receiver to force reverts
+			rejectAllIx, err := test_ccip_receiver.NewSetRejectAllInstruction(true, receiverTargetAccountPDA, deployer.PublicKey()).ValidateAndBuild()
+			require.NoError(t, err)
+			res := soltestutils.SendAndConfirm(ctx, t, solChains[destChain].Client, []solana.Instruction{rejectAllIx}, *deployer, solconfig.DefaultCommitment)
+			require.Nil(t, res.Meta.Err)
+		*/
+		/*
+			deployer := solChains[destChain].DeployerKey
+			// Set reject all flag in receiver to force reverts
+			test_ccip_receiver.SetProgramID(receiverProgram)
+			rejectAllIx, err := test_ccip_receiver.NewSetRejectAllInstruction(true, receiverTargetAccountPDA, deployer.PublicKey()).ValidateAndBuild()
+			require.NoError(t, err)
+			res := soltestutils.SendAndConfirm(ctx, t, solChains[destChain].Client, []solana.Instruction{rejectAllIx}, *deployer, solconfig.DefaultCommitment)
+			require.Nil(t, res.Meta.Err)
+		*/
+
+		out = mt.Run(
+			t,
+			mt.TestCase{
+				ValidationType:         mt.ValidationTypeCommit,
+				TestSetup:              setup,
+				Nonce:                  nil, // Solana nonce check is skipped
+				Receiver:               receiver,
+				MsgData:                bytes.Repeat([]byte("a"), 1),
+				ExtraArgs:              extraArgs,
+				NumberOfMessages:       2,
+				UseMulticall3:          true,
+				ExpectedExecutionState: testhelpers.EXECUTION_STATE_INPROGRESS,
+				//ExpectedExecutionState: testhelpers.EXECUTION_STATE_SUCCESS,
```

Comment:
The test utils do not support this. I'm using `mt.ValidationTypeCommit` in order to implement a custom validation outside of the testutil framework.